### PR TITLE
Run autoencoder experiment

### PR DESCRIPTION
**Pull request by scott-sievert-3**
**What does this PR implement?**
This PR adds functionality to run an autoencoder experiment in https://github.com/stsievert/adadamp-experiments/pull/4. Right now the changes are pretty minimal.

- [ ] check AdAdaGrad impl (in radadamp)
- [x] implement ResNet VAE (in adadamp-experiments)
- [x] fix git/train.py for autoencoder (here)

Specifically, this PR implements the following:

- [x] fixes RadaDamp to have the damping change like `1 / (loss / loss_0 + grad2 / grad2_0`
- [x] implements PadaDamp where damping changes like `ibs + updates*rate`

**Reference issues/PRs**
https://github.com/stsievert/adadamp-experiments/pull/5

### CODE REVIEW

**Comment by scott-sievert-3** [comment]
I think RadaDamp and PadaDamp work. At least in my brief tests, they both run and they both increase the batch size.

File: adadamp/damping.py [hunk 1]
```diff
@@ -411,63 +383,91 @@ def damping(self) -> int:
...
 
-    def __init__(self, *args, batch_growth_rate=None, **kwargs):
-        self.batch_growth_rate = batch_growth_rate
+    Parameters
+    ----------
+    rho : float (default=0.5)
+        Memory. High rho means slow adaptation, very stable. Low rho means very
+        adaptive, quick reaction.
+    """
+    def __init__(self, *args, reduction: str = "mean", rho: float = 0.5, **kwargs):
```
Comment (stsievert) [inline adadamp/damping.py:417]:
```suggestion
    def __init__(self, *args, reduction: str = "min", rho: float = 0.2, **kwargs):
```

File: adadamp/damping.py [hunk 2]
```diff
@@ -411,63 +383,91 @@ def damping(self) -> int:
...
+        if not (0 <= self.rho < 1):
+            raise ValueError(f"rho={self.rho} not valid, not in 0 <= rho < 1")
+
+        norm2_0, loss_0 = self._initial
+        self.norm2 = self.rho * self.norm2 + (1 - self.rho) * norm2
+        self.loss = self.rho * self.loss + (1 - self.rho) * loss
+        self._meta["norm2_hist"] = []
+        self._meta["loss_hist"] = []
+        bs = self.initial_batch_size / (0.5 * (self.loss / loss_0 + self.norm2 / norm2_0))
+        print(bs)
```
Comment (stsievert) [inline adadamp/damping.py:469]:
```suggestion
```

**Review by stsievert — APPROVED** [review]
Thanks for the contribution!
